### PR TITLE
background_job_sidekiq

### PR DESCRIPTION
**Pull request by manazirahsan**

### CODE REVIEW

File: app/views/tms_mailer/email_trainee_report.html.erb [hunk 1]
```diff
@@ -1 +1,23 @@
-Some custom message will be sent to supervisor about trainee's report
\ No newline at end of file
+<h4><%= t :greeting, :user_name => @user.name %></h4>
+<%= t :trainee_completed_tasks, :trainee_name => @user.name %>
+<div class="row">
+  <div class="col-md-8 col-md-offset-2">
+    <table class="table table-border table-striped">
+      <thead>
+      <tr>
+        <td>Task Name</td>
```
Comment (MuhammadTamzid) [inline app/views/tms_mailer/email_trainee_report.html.erb]: i18n

File: app/models/course.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Course < ActiveRecord::Base
+  scope :courses_to_be_sent_deadline, -> {where "created_at > :within_two_weeks", :within_two_weeks => Time.now - 14.days }
```
Comment (dieunb) [inline app/models/course.rb]: too long, and wrong convention

File: app/models/user_task.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class UserTask < ActiveRecord::Base
+  scope :tasks_completed_by_trainee_today, -> {where "created_at > :within_today", :within_today => Time.now - 1.day }
```
Comment (dieunb) [inline app/models/user_task.rb]: too long, convention

File: app/views/tms_mailer/email_about_deadline.html.erb [hunk 4]
```diff
@@ -1 +1,5 @@
-This is a place holder for the supervisors and the trainees about deadline of their courses.
\ No newline at end of file
+<h4><%= t :greeting, :user_name => @user.name %></h4>
```
Comment (dieunb) [inline app/views/tms_mailer/email_about_deadline.html.erb]: convention

File: app/views/tms_mailer/email_about_deadline.html.erb [hunk 5]
```diff
@@ -1 +1,5 @@
...
+<%= t :deadline_of_a_course, :course_name => @course.name, :course_deadline => @course.end_date %>
```
Comment (dieunb) [inline app/views/tms_mailer/email_about_deadline.html.erb]: convention

File: app/views/tms_mailer/email_trainee_report.html.erb [hunk 6]
```diff
@@ -1 +1,23 @@
-Some custom message will be sent to supervisor about trainee's report
\ No newline at end of file
+<h4><%= t :greeting, :user_name => @user.name %></h4>
```
Comment (dieunb) [inline app/views/tms_mailer/email_trainee_report.html.erb]: convention

File: config/initializers/redis.rb [hunk 7]
```diff
@@ -0,0 +1 @@
+$redis = Redis.new(:host => 'localhost', :port => 6379)
```
Comment (MuhammadTamzid) [inline config/initializers/redis.rb]: Double quote

Comment (dieunb) [inline config/initializers/redis.rb]: convention

File: app/views/tms_mailer/email_trainee_report.html.erb (hunk 1 above)
Comment (shakil-fg) [inline app/views/tms_mailer/email_trainee_report.html.erb]: i18n

**Comment by shakil-fg** [comment]
OK

File: config/initializers/redis.rb [hunk 8]
```diff
@@ -0,0 +1 @@
+$redis = Redis.new(:host => "localhost", :port => 6379)
```
Comment (dieunb) [inline config/initializers/redis.rb]: Convention

File: app/models/course.rb [hunk 9]
```diff
@@ -1,4 +1,7 @@
 class Course < ActiveRecord::Base
+  scope :courses_to_be_sent_deadline, -> {where "created_at > :within_two_weeks",
+    within_two_weeks: Time.now - 14.days }
```
Comment (dieunb) [inline app/models/course.rb]: `within_two_weeks: Time.now - 14.days}
`

File: app/models/user_task.rb [hunk 10]
```diff
@@ -1,4 +1,7 @@
 class UserTask < ActiveRecord::Base
+  scope :tasks_completed_by_trainee_today, -> {where "created_at > :within_today",
+    within_today: Time.now - 1.day }
```
Comment (dieunb) [inline app/models/user_task.rb]: `within_today: Time.now - 1.day}`

**Comment by noor-framgia** [comment]
ok

**Comment by mdmahmudurrahman** [comment]
Ok

**Pull request merged**